src/requests.rs

```diff
@@ -73,3 +78,39 @@ pub fn compile(client: Client, host: &str, src: String, compiler: &str, args: St
     }
     res
 }
+
+pub fn shorten(client: Client,
+               host: &str,
+               src: String,
+               compiler: &str,
+               args: String) -> String {
+    // TODO: `compile` and `shorten` behave differently, as I was not able to
```

Comment:
@ethanhs would you know how to pass the same data to `shorten`, so that the behavior is the same between this and `compile`?

Reply:
Perhaps have an intermediate function do the work, then dispatch to the correct one? What issue did you run into?

Reply:
I was referring to the API being different for the two: https://github.com/mattgodbolt/compiler-explorer/blob/master/docs/API.md 

Shorten takes a "ClientState" object as input ([1]), where as `../../compile` has a different JSON structure.

@ethanhs I'll update this PR in the next few days. Please excuse the delay.

[1]: https://github.com/mattgodbolt/compiler-explorer/blob/master/lib/clientstate.js